src/textual/widgets/_button.py

```diff
@@ -1,67 +1,75 @@
 from __future__ import annotations
 
-from rich.align import Align
-from rich.console import Console, ConsoleOptions, RenderResult, RenderableType
-from rich.style import StyleType
+from typing import cast
+
+from rich.console import RenderableType
+from rich.text import Text
 
 from .. import events
 from ..message import Message
 from ..reactive import Reactive
 from ..widget import Widget
 
 
-class ButtonPressed(Message, bubble=True):
-    pass
-
-
-class Expand:
-    def __init__(self, renderable: RenderableType) -> None:
-        self.renderable = renderable
-
-    def __rich_console__(
-        self, console: Console, options: ConsoleOptions
-    ) -> RenderResult:
-        width = options.max_width
-        height = options.height or 1
-        yield from console.render(
-            self.renderable, options.update_dimensions(width, height)
-        )
-
+class Button(Widget, can_focus=True):
+    """A simple clickable button."""
+
+    CSS = """
+    
+    Button {
+        width: auto;
+        height: 3;
+        padding: 0 2;
+        background: $primary;
+        color: $text-primary;
+        content-align: center middle;
+        border: tall $primary-lighten-3;        
+        margin: 1;
+        min-width:16;
```

Comment:
I'd maybe leave this to the user as well?

If I were to drop a button into my app and wanted it to be small e.g. a button with "Yes" or "No", I would need to know that I need to override the "min-width" property which isn't intuitive to me.

It also means I wouldn't be able to set the `width` property to anything below 16, right? It seems `min-width` takes precedence.

Reply:
Done. Although without a min-width, the buttons may be all different sizes which may not look good.